src/ledger/LedgerTxn.cpp

```diff
@@ -3455,55 +3461,84 @@ LedgerTxnRoot::Impl::getNewestVersion(InternalLedgerKey const& gkey) const
     }
     auto const& key = gkey.ledgerKey();
 
-    if (mEntryCache.exists(key))
-    {
-        std::string zoneTxt("hit");
-        ZoneText(zoneTxt.c_str(), zoneTxt.size());
-        return getFromEntryCache(key);
-    }
-    else
-    {
-        std::string zoneTxt("miss");
-        ZoneText(zoneTxt.c_str(), zoneTxt.size());
-        ++mPrefetchMisses;
-    }
-
-    std::shared_ptr<LedgerEntry const> entry;
-    try
+    // If everything is cached in-memory, don't check or populate the cache
+    if (mApp.getConfig().BUCKETLIST_DB_INDEX_PAGE_SIZE_EXPONENT == 0)
     {
-        if (key.type() != OFFER)
+        std::shared_ptr<LedgerEntry const> entry;
+        try
         {
             entry = getSearchableLiveBucketListSnapshot().load(key);
         }
-        else
+        catch (std::exception& e)
```

Comment:
for future-proofing, should we keep all three catch clauses here (or at least catch(...) )? We have to abort in this critical path in case of a failure, so safer to catch everything in case the underlying DB load path start throwing custom exceptions.

Reply:
I added back the catch all, I don't think SOCI related catches make sense since we're not hitting SQL in this path.

Reply:
Sg. Actually, `NonSociRelatedException` is just dead code now that all the *SQL files have been removed. While we're already here, would you mind cleaning that up? (we can just delete NonSociRelatedException.h/cpp)

Reply:
Done